Documentation/contributors/design-proposals/installed-api.md

```diff
@@ -0,0 +1,282 @@
+    No resources found.
+    ```
+
+2. Watching changes to the list of installed operators.
+
+    Ex. Using `kubectl` to open a watch on installed operators.
+
+    ```sh
+    # An installed operator in namespace default transitions from Succeeded to Failed
+    $ kubectl -n default get installed.packages.operators.coreos.com -w
+    NAME                              INSTALLATION_NAMESPACE   CHANNEL             CURRENT_VERSION   TARGET_VERSION   PHASE
+    etcdoperator.v0.9.4-clusterwide   operators                clusterwide-alpha   0.9.4             0.9.4            Succeeded
+    etcdoperator.v0.9.4-clusterwide   operators                clusterwide-alpha   0.9.4             0.9.4            Failed
+    ```
+
+### Non-Use Cases
+
+1. External[^1] clients may not create, delete, or mutate `installed.packages.operators.coreos.com` resources.
+
+    Ex. Unable to create an `Installed` resource with `kubectl`
+
+    ```sh
+    # Attempt to create Installed resource in namespace mojave
+    $ cat <<EOF | kubectl -n mojave create -f -
+    apiVersion: v1alpha2/packages.operators.coreos.com
+    kind: Installed
+    # <continued resource definition>
+    EOF
+    Error from server (MethodNotAllowed): error when creating "STDIN": the server does not allow this method on the requested resource
+    ```
+
+[^1]: Clients external to the API implementation (e.g. not part of the controller responsible for them).
+
+### Supported HTTP Verbs
+
+`installed.packages.operators.coreos.com` HTTP verb support table:
+
+|   Verb   | Supported |
+|:--------:|:---------:|
+|  `GET`   |     ✔     |
+|  `PUT`   |     ❌     |
+|  `POST`  |     ❌     |
+| `PATCH`  |     ❌     |
+| `DELETE` |     ❌     |
+
+## Implementation
+
+### Aggregated API
+
+To circumvent the overhead of managing copies of persistent resources in each namespace, like `PackageManifests`, `Installed` resources will be _synthesized_ on request by an extension API server . The server will watch tracked resources and cache generated `Installed` instances for each operator (`CSV`) on the cluster. When a `GET` is received, the server will retrieve the cached instances configured to watch the request namespace and set `metadata.namespace` to match it before returning the results.
+
+The `Installed` resource belongs to the same API group (`packages.operators.coreos.com`) as `PackageManifest`, but at different versions (`v2alpha1` and `v1` respectively). It follows that they should share the internal types and conversion logic, which makes `packageserver` an ideal place to add this API. However, due to [current limitations in OLM](https://github.com/operator-framework/operator-lifecycle-manager/issues/727) a separate `Deployment` must be introduced into the `packageserver` `CSV` to support the new version (`v2alpha1`). To this end, the names of each deployment will be of the format `packageserver.<version>` and will be referenced in their respective `owned` entry:
```

Comment:
I’m not against putting this in packageserver but I don’t really see any particular reason to. 

I don’t believe there’s any shared functionality between this proposed apiserver and the packageserver?

Reply:
If they share an API group, it behooves us from an internal types/codegen perspective to have versioned structures exist in the same go package. The servers themselves could be separate and live in separate packages.

Reply:
Why should they or should they not share an apigroup?